telluric/vrt.py

```
@@ -239,11 +240,17 @@ def max_resolution():
             xsize = raster.width * raster.affine.a / affine.a
             ysize = raster.height * raster.affine.e / affine.e
             dst_window = Window(xoff, yoff, xsize, ysize)
-            file_name = raster._filename if relative_to_vrt else os.path.join(os.getcwd(), raster._filename)
+            file_name = raster.source_file if relative_to_vrt else os.path.join(os.getcwd(), raster.source_file)
+
             vrt.add_band_simplesource(band_element, band_idx, raster.dtype, relative_to_vrt, file_name,
                                       raster.width, raster.height,
                                       raster.block_shape(i)[1], raster.block_shape(i)[0],
                                       src_window, dst_window)
+            if i == 0:
```

you are assuming here that always the mask is from the first raster. can you use an argument that the user can set which raster should be used as mask and `None` for no mask at all?

---

what should be the default?

I think I will do something else:

I will look for this flag: https://github.com/mapbox/rasterio/blob/master/rasterio/enums.py#L81

and then it doesn't matter which band I use, 

If it is all_valid we don't need mask at all
if it is no data I will set the no-data value
if it alpha I will set the alpah band

WDYT

---

good idea, but you can have different flags for different rasters, and you still need to deside which of the rasters is the "leading one"
regarding the default, I think it should be `None` (no mask) and for any other case the user should pick the raster he prefers.